apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -298,6 +296,79 @@ defmodule GameBackend.CurseOfMirra.Quests do
     |> Repo.transaction()
   end
 
+  def get_user_quest_progress(%UserQuest{quest: %Quest{} = quest} = user_quest, arena_match_results) do
+    arena_match_results =
+      case quest.type do
+        type when type in ["daily", :daily] ->
+          Enum.filter(arena_match_results, fn arena_match_result ->
+            user_quest.activated_at &&
+              NaiveDateTime.compare(arena_match_result.inserted_at, user_quest.activated_at) == :gt &&
+              NaiveDateTime.diff(arena_match_result.inserted_at, user_quest.inserted_at, :day) == 0
+          end)
+
+        _ ->
+          arena_match_results
+      end
+
+    arena_match_results
+    |> filter_results_that_meet_quest_conditions(quest.conditions)
+    |> Enum.reduce(0, fn arena_match_result, acc ->
+      type = arena_match_result_field_to_atom(quest.objective["match_tracking_field"])
+      # We won't accumulate progress if the objective has a wrong field getter
+      accumulate_objective_progress_by_scope(quest.objective["scope"], Map.get(arena_match_result, type))
+      |> case do
+        nil ->
+          acc
+
+        progress ->
+          acc + progress
+      end
+    end)
+  end
+
+  def try_to_complete_quest_for_user(user, user_quest) do
```

Comment:
```suggestion
  def complete_user_quest(user, user_quest) do
```

is a better name, there's no try. If it fails it will return error.

Reply:
completly agreed